catalog/ui/catalog-ui-search/src/main/webapp/component/search-settings/search-settings.view.js

```diff
@@ -0,0 +1,92 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define, setTimeout*/
+define([
+    'marionette',
+    'underscore',
+    'properties',
+    'jquery',
+    './search-settings.hbs',
+    'js/CustomElements',
+    'component/singletons/user-instance',
+    'component/property/property.view',
+    'component/property/property'
+], function (Marionette, _, properties, $, template, CustomElements, user, PropertyView, Property) {
+
+    return Marionette.LayoutView.extend({
+        template: template,
+        tagName: CustomElements.register('search-settings'),
+        modelEvents: {},
+        events: {
+            'click .editor-edit': 'turnOnEditing',
+            'click .editor-cancel': 'cancel',
+            'click .editor-save': 'save'
+        },
+        regions: {
+            propertyResultCount: '.property-result-count .property-region'
+        },
+        onBeforeShow: function () {
+            this.setupResultCount();
+            this.startListeningToRegions();
+            this.turnOnEditing();
+        },
+        startListeningToRegions: function(){
+            this.propertyResultCount.currentView.$el.on('change', this.save.bind(this));
+        },
+        stopListeningToRegions: function(){
+            this.stopListening(this.propertyResultCount.currentView.$el);
+        },
+        setupResultCount: function () {
+            var resultCount = properties.resultCount;
+            var userResultCount = user.get('user').get('preferences').get('resultCount');
+
+            if (userResultCount !== undefined){
+                resultCount = userResultCount;
+            }
+
+            this.propertyResultCount.show(new PropertyView({
+                model: new Property({
+                    value: [resultCount],
+                    min: 1,
+                    max: properties.resultCount,
+                    type: 'INTEGER'
+                })
+            }));
+
+            this.propertyResultCount.currentView.turnOnLimitedWidth();
+        },
+        turnOnEditing: function () {
+            this.$el.addClass('is-editing');
+            this.regionManager.forEach(function (region) {
+                if (region.currentView) {
+                    region.currentView.turnOnEditing();
+                }
+            });
+        },
+        save: function () {
+            var preferences = user.get('user').get('preferences');
+            var userProvidedCount = this.propertyResultCount.currentView.getCurrentValue()[0];
+            var resultCountValue = Math.min(properties.resultCount, userProvidedCount);
+            resultCountValue = Math.max(1, resultCountValue);
+
+            this.propertyResultCount.currentView.setValue(resultCountValue);
```

Comment:
Earlier reviewers have touched on it, but I think we need to be a bit more explicit to the user about what's happening with the bounds here.